Timesheet.gs

```diff
@@ -67,134 +74,152 @@ function checkOut (arg1, arg2, properties, lock) {return eventHandler_(EVENT_HAN
   
-    Assert.handleError(error, config[2], Log_)
-    
-  } finally {
-  
-    if (!originallyHasLock) {
-      lock.releaseLock()
+    var handleError = Assert.HandleError.DISPLAY_FULL
+
+    if (!PRODUCTION_VERSION_) {
+      handleError = Assert.HandleError.THROW
     }
-  }
-  
-  return
-  
-  // Private Functions
-  // -----------------
-
-  /**
-   * Initialise the event handling
-   */
- 
-  function initialseEventHandler() {
-      
-    var userEmail = Session.getEffectiveUser().getEmail()
 
-    Assert.init({
-      handleError:    HANDLE_ERROR_, 
+    var assertConfig = {
+      error:          error,
+      userMessage:    config[1],
+      log:            Log_,
+      handleError:    handleError, 
       sendErrorEmail: SEND_ERROR_EMAIL_, 
-      emailAddress:   ADMIN_EMAIL_ADDRESS_ + ',' + userEmail,
+      emailAddress:   ADMIN_EMAIL_ADDRESS_,
       scriptName:     SCRIPT_NAME,
       scriptVersion:  SCRIPT_VERSION, 
-    })
+    }
 
-    if (PRODUCTION_VERSION_) {
+    Assert.handleError(assertConfig) 
+  }
+  
+} // eventHandler_()
+
+// Private event handlers
+// ----------------------
+
+function onInstall_(event){
+
+  var triggerOpenId = properties.getProperty('triggerOpenId')
+  
+  if (triggerOpenId !== null) {
+    properties.deleteProperty('triggerOpenId')
     
-      var firebaseUrl = properties.getProperty(PROPERTY_FIREBASE_URL)
-      var firebaseSecret = properties.getProperty(PROPERTY_FIREBASE_SECRET)
+  }
+    
+  onOpen_(event)
+}
 
-      Log_ = BBLog.getLog({
-        displayUserId:        BBLog.DisplayUserId.USER_KEY_FULL,
-        lock:                 lock,
-        firebaseUrl:          firebaseUrl,
-        firebaseSecret:       firebaseSecret,
-      });
+function onOpen_(event) {
     
+  var menu = SpreadsheetApp
+        .getUi().createMenu('[ Timesheet ]')
+        
+      menu.addItem('Check In',  'checkIn')
+      menu.addItem('Check Out', 'checkOut')
+        
+  if (event.authMode === ScriptApp.AuthMode.NONE) {
+   
+    menu.addItem('Start', 'initialize') 
+
+  } else { // LIMITED or FULL
+  
+    var triggerOpenId = PropertiesService.getDocumentProperties().getProperty('triggerOpenId')
+        
+    if (triggerOpenId === null) {
+  
+      menu.addItem('Start', 'initialize')   
+  
     } else {
+    
+      var html = HtmlService
+      .createHtmlOutputFromFile("index")
+      .setTitle("Check In/Check Out");
+      SpreadsheetApp.getUi().showSidebar(html)
+    }    
+  }
+  menu.addToUi()
+}
 
-      Log_ = BBLog.getLog({
-        level:                DEBUG_LOG_LEVEL_, 
-        displayFunctionNames: DEBUG_LOG_DISPLAY_FUNCTION_NAMES_,
-        lock:                 lock,
-      })
-    }
+function isCheckedIn_(properties) {
 
-  } // eventHandler_.initialseEventHandler() 
+  var status = properties.getProperty("TIMESHEET_STATUS")
+  Log_.fine('status: ' + status)
 
-} // eventHandler_()
+  if (status === "CHECKED_IN"){
+    return true
+  } else {
+    return false
+  }
```

Comment:
tip: This could be 

    return (status === 'CHECKED_IN' ? true : false

Reply:
Changed